astropy/io/fits/hdu/hdulist.py

```diff
@@ -694,14 +694,13 @@ def writeto(self, fileobj, output_verify='exception', overwrite=False,
         # but only if the file doesn't exist.
         fileobj = _File(fileobj, mode='ostream', overwrite=overwrite)
         hdulist = self.fromfile(fileobj)
+        dirname = os.path.dirname(hdulist._file.name)
```

Comment:
OK, I've added it back. However not sure whether that case would make any sense here, if dirname is ``None``, it will try to get the directory name from ``self._file.name``, which may not exist either. Anyway, tests pass both ways, so any problem will be an extreme corner case that only comes up when there isn't enough space left on disk :)